system-helper/flatpak-system-helper.c

```diff
@@ -141,6 +205,123 @@ no_progress_cb (OstreeAsyncProgress *progress, gpointer user_data)
 {
 }
 
+static gboolean
+handle_create_pull_cache_dir (FlatpakSystemHelper   *object,
+                              GDBusMethodInvocation *invocation,
+                              const gchar           *arg_ref,
+                              const gchar           *optional_commit,
+                              const gchar           *arg_installation)
+{
+  GDBusConnection *connection = g_dbus_method_invocation_get_connection (invocation);
+  const gchar *sender = g_dbus_method_invocation_get_sender (invocation);
+  FlatpakSystemHelperOngoingPull *new_pull;
+  g_autoptr(FlatpakDir) system = NULL;
+  g_autoptr(GError) error = NULL;
+  g_autoptr(GDBusMessage) msg = NULL;
+  g_autoptr(GDBusMessage) reply = NULL;
+  g_autofree gchar *cache_tmpdir = NULL;
+  g_autofree gchar *user_folder = NULL;
+  g_autofree gchar *root_folder = NULL;
+  g_autofree gchar *flatpak_dir = NULL;
+  guint32 uid;
+  guint uid_pull_count = 0;
+
+  system = dir_get_system (arg_installation, &error);
+  if (system == NULL)
+    {
+      g_dbus_method_invocation_return_gerror (invocation, error);
+      return TRUE;
+    }
+
+  if (ongoing_pulls == NULL)
+    ongoing_pulls = g_ptr_array_new ();
+
+  msg = g_dbus_message_new_method_call ("org.freedesktop.DBus",
+                                        "/org/freedesktop/DBus",
+                                        "org.freedesktop.DBus",
+                                        "GetConnectionCredentials");
+  g_dbus_message_set_body (msg, g_variant_new ("(s)", sender));
+
+  reply = g_dbus_connection_send_message_with_reply_sync (connection, msg,
+                                                          G_DBUS_SEND_MESSAGE_FLAGS_NONE,
+                                                          -1,
+                                                          NULL,
+                                                          NULL,
+                                                          &error);
+  if (reply == NULL)
+    return FALSE;
+
+  if (g_dbus_message_get_message_type (reply) == G_DBUS_MESSAGE_TYPE_METHOD_RETURN)
+    {
+      GVariant *body = g_dbus_message_get_body (reply);
+      g_autoptr(GVariantIter) iter = NULL;
+      const char *key;
+      GVariant *value;
+
+      g_variant_get (body, "(a{sv})", &iter);
+      while (g_variant_iter_loop (iter, "{&sv}", &key, &value))
+        {
+          if (strcmp (key, "UnixUserID") == 0)
+            {
+              uid = g_variant_get_uint32 (value);
+              break;
+            }
+        }
+    }
+
+  for (guint i = 0; i < ongoing_pulls->len; i++)
+    {
+      FlatpakSystemHelperOngoingPull *pull_tmp;
+
+      pull_tmp = g_ptr_array_index (ongoing_pulls, i);
+      if (pull_tmp->uid = uid)
+        {
+          uid_pull_count++;
+          /*
+          * If there is a already a partial pull available, return that.
+           */
```

Comment:
Is this a comment describing the code in its current state, or describing something which is still to be implemented? If the latter, I suggest prefixing it with `TODO: ` so that it’s clear to you (via `git grep`) what still needs to be done; and it’s clear to any reviewers too.